src/frontend/utils/conversejs/converse-plugins/nicknameManagementPlugin.ts

```diff
@@ -9,17 +10,36 @@ const addNicknameManagementPlugin = (xmpp: XMPP) =>
     initialize() {
       const _converse = this._converse;
 
-      const claimNewNicknameInChatRoom = (newNickname: string): void => {
+      const claimNewNicknameInChatRoom = (
+        newNickname: string,
+        callbackSuccess: (stanza: HTMLElement) => void,
+        callbackError: (stanza: HTMLElement) => void,
+      ): void => {
+        const completeCallbackSuccess = (stanza: HTMLElement) => {
+          callbackSuccess(stanza);
+          _converse.api.settings.set('nickname', newNickname);
+        };
         const presence = converse.env.$pres({
           from: _converse.connection.jid,
           to: xmpp.conference_url + '/' + newNickname,
         });
-        _converse.connection.send(presence);
+        _converse.connection.sendPresence(
+          presence,
+          completeCallbackSuccess,
+          callbackError,
+          PROSODY_TIMEOUT_ON_NICKNAME_CHANGE_REQUEST_IN_MS,
+        );
       };
 
       Object.assign(converse, {
         claimNewNicknameInChatRoom,
       });
+
+      const resetConnection = () => _converse.initConnection();
+
+      Object.assign(converse, {
+        resetConnection,
+      });
```

Comment:
I don't see where you are using it and you can assign only once to converse both `claimNewNicknameInChatRoom` and `resetConnection`

Reply:
Yes, i forgot to delete it (I was using it in a testing purpose)